src/Microsoft.Extensions.Logging/Logger.cs

```diff
@@ -197,5 +215,87 @@ public void Dispose()
                 }
             }
         }
+
+        private class Metric : IMetric
+        {
+            private readonly string _name;
+
+            private Logger _logger;
+            private IMetric[] _metrics;
+            private object _sync = new object();
+
+            public Metric(Logger logger, string name)
+            {
+                _logger = logger;
+                _name = name;
+            }
+
+            public void RecordValue(double value)
```

Comment:
As we discussed this implementation has to be optimized to update metrics array only when required.